src/exec/no_pty.rs

```diff
@@ -56,39 +59,13 @@ pub(super) fn exec_no_pty(sudo_pid: ProcessId, mut command: Command) -> io::Resu
         err
     })?
     else {
-        // Restore the signal mask now that the handlers have been setup.
-        if let Some(set) = original_set {
-            if let Err(err) = set.set_mask() {
-                dev_warn!("cannot restore signal mask: {err}");
-            }
-        }
-
-        // SAFETY: We immediately exec after this call and if the exec fails we only access stderr
-        // and errpipe before exiting without running atexit handlers using _exit
-        if let Err(err) = unsafe { file_closer.close_the_universe() } {
-            dev_warn!("failed to close the universe: {err}");
-            // Send the error to the monitor using the pipe.
-            if let Some(error_code) = err.raw_os_error() {
-                errpipe_tx.write(&error_code).ok();
-            }
-
-            // We call `_exit` instead of `exit` to avoid flushing the parent's IO streams by accident.
-            _exit(1);
-        }
-
-        let err = command.exec();
-
-        dev_warn!("failed to execute command: {err}");
-        // If `exec` returns, it means that executing the command failed. Send the error to the
-        // monitor using the pipe.
-        if let Some(error_code) = err.raw_os_error() {
-            errpipe_tx.write(&error_code).ok();
-        }
-
-        // We call `_exit` instead of `exit` to avoid flushing the parent's IO streams by accident.
-        _exit(1);
+        exec_command(file_closer, command, original_set, errpipe_tx);
     };
 
+    if let Some(spawner) = spawn_noexec_handler {
```

Comment:
I investigated a bit whether this caused a race condition; but that is not the case since the `seccomp` filter has already been applied even if the thread hasn't spawned yet. And even if a child process hits the seccomp filter before the thread is ready, it will at that point simply block until the thread *has* come online, can you confirm that? (E.g. by resolving this comment) 

For the `pty` case there is no potential for a race because of the "green light" mechanism, so this was the only place where I put some extra attention during review.

I do think moving it in the event loop in the future still might be a good idea, but for now using a thread makes for easier maintainability, and we simply don't have the time to fine tine this further. I'll park it as an issue.